cw/util/device.py

```diff
@@ -158,7 +158,16 @@ def initialize_scope(self, scope_gain, num_samples, offset, pll_frequency):
         if not husky:
             scope.clock.reset_adc()
             time.sleep(0.5)
-        assert (scope.clock.adc_locked), "ADC failed to lock"
+
+        # Wait for ADC to lock.
+        ping_cnt = 0
+        while not scope.clock.adc_locked:
+            if ping_cnt == 3:
+                raise RuntimeError(
+                    f'"ADC failed to lock" (attempts: {ping_cnt}).')
```

Comment:
nit: double quotes seem unnecessary
```suggestion
                    f'ADC failed to lock (attempts: {ping_cnt}).')
```

Reply:
Done, thanks @andreaskurth !